atomix/cluster/src/main/java/io/atomix/raft/cluster/impl/RaftClusterContext.java

```diff
@@ -208,72 +164,142 @@ private void createInitialConfig(final Collection<MemberId> cluster) {
 
     // Create a new configuration and store it on disk to ensure the cluster can fall back to the
     // configuration.
-    configure(new Configuration(0, 0, localMember.getLastUpdated().toEpochMilli(), activeMembers));
+    final var initialConfiguration =
+        new Configuration(0, 0, localMember.getLastUpdated().toEpochMilli(), activeMembers);
+    configure(initialConfiguration);
+    commitCurrentConfiguration();
+  }
+
+  /** Returns the context for a given member. */
+  public RaftMemberContext getMemberContext(final MemberId id) {
+    return remoteMemberContexts.get(id);
   }
 
   /**
-   * Returns a member by ID.
+   * Calculates the smallest value that is reported for a majority of this cluster, assuming that
+   * the local node always has the highest value.
    *
-   * @param id The member ID.
-   * @return The member.
+   * @param calculateMemberValue a function that calculates a value for a given member. Will be
+   *     evaluated at least once for every remote member.
+   * @return empty when no remote members are present, otherwise the smallest value that is reported
+   *     by enough remote members to form a quorum with the local member.
    */
-  public DefaultRaftMember getRemoteMember(final MemberId id) {
-    final RaftMemberContext member = membersMap.get(id);
-    return member != null ? member.getMember() : null;
+  public <T extends Comparable<T>> Optional<T> getQuorumFor(
+      final Function<RaftMemberContext, T> calculateMemberValue) {
+    final var contexts = new ArrayList<>(remoteActiveMembers);
+
+    if (configuration.requiresJointConsensus()) {
+      final var oldMembers = configuration.oldMembers();
+      final var newMembers = configuration.newMembers();
+
+      final var oldContexts =
+          contexts.stream()
+              .filter(context -> oldMembers.contains(context.getMember()))
+              .collect(Collectors.toCollection(ArrayList::new));
+      final var newContexts =
+          contexts.stream()
+              .filter(context -> newMembers.contains(context.getMember()))
+              .collect(Collectors.toCollection(ArrayList::new));
+
+      final var oldQuorum = getQuorumFor(oldContexts, calculateMemberValue);
+      final var newQuorum = getQuorumFor(newContexts, calculateMemberValue);
+      if (oldQuorum.isPresent() && newQuorum.isPresent()) {
```

Comment:
:wrench: Actually this always returns true. oldQuorum and newQuorum is never empty. Please check it and remove the `else` case if it applies.

Reply:
:+1: My IDE was showing warning that this is never empty. If this is not true, then no problem.